src/libgit2/object.c

```diff
@@ -520,13 +520,38 @@ int git_object_lookup_bypath(
 	return error;
 }
 
+int git_object__abbrev_length(int *out, git_repository *repo)
```

Comment:
I think that probably this should be in the `git_repository` group, instead of `git_object`. (It's really about the repository's abbreviation settings.) This is a minor nit, and I've kept you waiting on a review for a long time, so I'm happy to make this change later today if you prefer. 🙏 

Thanks for investigating and fixing.

Reply:
Will move to git_repository__abbrev_length. I just happened to run into this issue and used it to get more practice with C, no rush :)